app/src/main/java/com/github/fribourgsdp/radio/SettingsActivity.kt

```diff
@@ -1,11 +1,112 @@
 package com.github.fribourgsdp.radio
 
+import android.content.res.Configuration
+import android.content.res.Resources
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.DisplayMetrics
+import android.util.Log
+import android.view.View
+import android.widget.*
+import android.content.Intent
+import java.util.Locale
+import android.widget.Toast
+
+import android.widget.AdapterView
+import android.os.Build
+
+
+
+
+
+
+
+
+
+
+
+class SettingsActivity : MyAppCompatActivity() {
+
+
+    private lateinit var spinnerLanguage: Spinner
+    private lateinit var saveSettingsButton : Button
+    private lateinit var settings : Settings
+    private lateinit var settingsBuilder : SettingsBuilder
+
 
-class SettingsActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         setContentView(R.layout.activity_settings)
+        saveSettingsButton = findViewById(R.id.approveButton)
+
+        Settings.loadOrDefault(this).addOnSuccessListener { settings ->
+            this.settings = settings
+            settingsBuilder = SettingsBuilder(settings)
+            initLanguageSpinner(settings.getLanguage())
+        }
+
+        initSaveSettingsButton()
+
+
     }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
+        val intent = Intent(this, MainActivity::class.java)
+        startActivity(intent)
+        finish()
+    }
+
+
+    private fun initSaveSettingsButton(){
+
+        saveSettingsButton.setOnClickListener{
+            settings = settingsBuilder.build()
+            settings.save(this)
+        }
+    }
+
+    private fun initLanguageSpinner(actualLanguage : Language){
+
+
+        spinnerLanguage = findViewById(R.id.spinner_language)
+        val languageList: Array<Language> =  Language.values()
+
+        val adapter: ArrayAdapter<Language> = ArrayAdapter<Language>(this,android.R.layout.simple_spinner_item,languageList)
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinnerLanguage.setAdapter(adapter);
+        spinnerLanguage.setSelection(actualLanguage.id, false);
+        spinnerLanguage.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>?,view: View?,position: Int,id: Long) {
+                if (parent != null && view != null) {
+                    onItemSelectedHandler(parent, view, position, id)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+    }
+
+    private fun onItemSelectedHandler(adapterView: AdapterView<*>, view: View, position: Int, id: Long) {
+        val adapter: Adapter = adapterView.adapter
+        val language: Language = adapter.getItem(position) as Language
+        Toast.makeText(this@SettingsActivity, "Selected Employee: $language",Toast.LENGTH_SHORT).show()
```

Comment:
I don't think Employee is what you meant.

Reply:
Yes, good point